### PR TITLE
Remove "galaxy" from Cromwell App's k8s resource names [BW-860]

### PR DESCRIPTION
**Pull request by cjllanwarne**
Cromwell's k8s resources are now functions of the cromwell prefixes in the database, rather than being hard-coded to use "galaxy"s configuration.

Note that this involved a little bit of refactoring of the app config code and identification of some common fields.

### CODE REVIEW

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/config/GkeAppConfig.scala
Comment (cahrens) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/config/GkeAppConfig.scala:38]: What does this do?

**Review by cahrens — APPROVED** [review]
Just the one question (because I'm not understanding the syntax… not because it's wrong).

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/config/GkeAppConfig.scala
Comment (cjllanwarne) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/config/GkeAppConfig.scala:38]: I guess the simple answer is, brings something that was happening somewhere else into this class.

Syntax wise, it's a lazy val (so gets evaluated the first time it's asked for, then remembered), which takes the services section from config and turns each of them into a KubernetesService instance, with an ID of `-1`. 

In terms of why, I'd defer to an IA member, but I guess implicitly it assumes that Galaxy and Cromwell configs only have a single service (at least in the leo config), otherwise the IDs would clash. And later on, this is used to calculate the proxy URLs

Comment (cahrens) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/config/GkeAppConfig.scala:38]: The ID of -1 was particularly confusing to me-- I'd love to hear more about that from IA folks.

Comment (Qi77Qi) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/config/GkeAppConfig.scala:38]: yea a bit confusing, but this is just a helper function for persisting `KubernetesService` to DB...`id` is an auto-incrementing key, and the way slick works is you just stub out the id when saving it to DB

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/config/Config.scala
Comment (Qi77Qi) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/config/Config.scala]: you might have some merge conflicts with https://github.com/DataBiosphere/leonardo/pull/2354

**Review by Qi77Qi — APPROVED** [review]

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/config/Config.scala
Comment (cahrens) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/config/Config.scala]: Definitely will have merge conflicts! I figured I would do the rebase/conflict resolution of this branch after mine merges (@cjllanwarne is out until Wednesday).

**Comment by cahrens** [comment]
@Qi77Qi @cjllanwarne FYI that I have rebased this branch on develop and resolved conflicts introduced by https://github.com/DataBiosphere/leonardo/pull/2354. The branch has been force pushed.

Testing notes: I first pushed up the changes on a new branch (https://github.com/DataBiosphere/leonardo/pull/2361) and verified that tests passed. I then pushed the updated JAR to my FIAB and tested deploying both Galaxy and Cromwell to make sure that the names look correct.

image

**Comment by cjllanwarne** [comment]
Subsequent changes to the PR look good to me 👍

**Pull request merged**
